### PR TITLE
Change value tracking to verify a parameter is used when mapping from that parameter to it's callers

### PR DESCRIPTION
**Pull request by ToddGrun**
Change the value tracking code such that parameters only add callers that specify that parameter as an argument.

Fixes https://github.com/dotnet/roslyn/issues/66771

### CODE REVIEW

File: src/Features/Core/Portable/ValueTracking/ValueTracker.FindReferencesProgress.cs [hunk 1]
```diff
@@ -172,22 +173,43 @@ private async Task TrackMethodInvocationArgumentsAsync(ReferenceLocation referen
...
+                {
+                    if (syntaxFacts.IsArgument(descendantNode))
+                    {
+                        // Ensure this is an argument to the requested invocation
+                        if (invocationSyntax != descendantNode.FirstAncestorOrSelf<SyntaxNode>(syntaxFacts.IsInvocationExpression))
+                            continue;
+
+                        var argumentOperation = semanticModel.GetOperation(descendantNode, cancellationToken);
+                        if (argumentOperation is IArgumentOperation { Parameter: { } argumentParameter } &&
+                            _parameterSymbol.Locations.SequenceEqual(argumentParameter.Locations, arg: default(VoidResult), (loc1, loc2, _) => loc1 == loc2))
```
Comment (ToddGrun) [inline src/Features/Core/Portable/ValueTracking/ValueTracker.FindReferencesProgress.cs:203]: >_parameterSymbol.Locations.SequenceEqual(argumentParameter.Locations, arg: default(VoidResult), (loc1, loc2, _) => loc1 == loc2) [](http://example.com/codeflow?start=28&length=128)

There has to be a better way to verify the symbols are the same, but the symbol equality checks were all failing.

Comment (ToddGrun) [reply src/Features/Core/Portable/ValueTracking/ValueTracker.FindReferencesProgress.cs:203]: Hrm, maybe I did something wrong before, I just tried it and it worked that time. Let me test this a bit more.

Comment (ToddGrun) [reply src/Features/Core/Portable/ValueTracking/ValueTracker.FindReferencesProgress.cs]: Yep, seems to work now. Good, I guess.

File: src/Features/Core/Portable/ValueTracking/ValueTracker.FindReferencesProgress.cs [hunk 2]
```diff
@@ -172,22 +172,43 @@ private async Task TrackMethodInvocationArgumentsAsync(ReferenceLocation referen
...
+            {
+                // Filter out invocations not containing an argument matching _parameterSymbol
+                var containsMatchingArg = false;
+                var invocationDescendants = invocationSyntax.DescendantNodes();
+                foreach (var descendantNode in invocationDescendants)
+                {
+                    if (syntaxFacts.IsArgument(descendantNode))
+                    {
+                        // Ensure this is an argument to the requested invocation
+                        if (invocationSyntax != descendantNode.FirstAncestorOrSelf<SyntaxNode>(syntaxFacts.IsInvocationExpression))
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/ValueTracking/ValueTracker.FindReferencesProgress.cs:197]: this isn't a great way to do things.  consider using ISyntaxFacts which will letyou decompose things in a language agnostic fashino, while still being:

1. somewhat strongly typed.
2. ensuring you don't have code that is waslking down infinitely deep and all the way back up again.

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/ValueTracking/ValueTracker.FindReferencesProgress.cs:197]: another approach is to simply use IOp to get the IInvocationOp for your invocation.  Then find the IArg corresponding to your parameter.  And then just see if it is .Implicit or not.

Comment (ToddGrun) [reply src/Features/Core/Portable/ValueTracking/ValueTracker.FindReferencesProgress.cs]: Ah, I was making this too complicated. As you mentioned IInvocationOp already has what I need.

**Review by CyrusNajmabadi — APPROVED** [review]

**Pull request merged**